File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    float alpha = count_letters(text);

    float words = count_words(text) + 1;

    float sentences = count_sentences(text);

    float letters = (alpha / words * 100);
    float savg = (sentences / words * 100);

    //printf("length %f words %f\n", letters, savg);
    //printf("alpha %f, words %f, sentences %f\n", alpha, words, sentences);

    float grade_level = 0.0588 * letters - 0.296 * savg - 15.8;

    if (grade_level >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade_level <= 1)
    {
        printf("Before Grade 1\n");
    }

    //uses round funtion call to round to the nearest int and not round down
    else if (grade_level <= 15)
    {
        printf("Grade %i\n", (int) round(grade_level));
    }
   // printf("%f alpha, %f words, %f sents", alpha, words, sentences);
}

int count_letters(string text)
{
    int n = strlen(text);
    int alpha = 0;

    for (int i = 0; i < n; i++)
    {
        if isalpha(text[i])
        {
            alpha++;
        }
    }
    return alpha;
}

//counts words
int count_words(string text)
{
    int n = strlen(text);
    int spaces = 0;

    for (int i = 0; i < n; i++)
    {
        if isspace(text[i])
        {
            spaces++;
        }
    }
    return spaces;
}

//counts sentences
int count_sentences(string text)
{
    int n = strlen(text);
    int sent = 0;

    for (int i = 0; i < n; i++)
    {
        if (text[i] == 46)
        {
            sent++;
        }
        if (text[i] == 33)
        {
            sent++;
        }
        if (text[i] == 63)
        {
            sent++;
        }
    }
    return sent;
}

File: week6/pset/sentimental-readability/readability.py
import cs50
import re


def main():
    text = input("text: ")

    # calls functions to get alpha chars, word count, and sentence count
    alpha = float(count_letters(text))
    words = float(count_words(text))
    sentences = float(count_sent(text))

    # values for readability equation
    letters = float(alpha / words * 100)
    sent_avg = float(sentences / words * 100)

    grade_lev = float(0.0588 * letters - 0.296 * sent_avg - 15.8)

    # print statements for specified conditions
    if (grade_lev >= 16):
        print("Grade 16+")
    elif (grade_lev <= 1):
        print("Before Grade 1")
    elif (grade_lev <= 15):
        print(f"Grade {round(grade_lev)}")


def count_letters(text):
    # function takes text from user and gives a count of characters in the text

    n = len(text)
    alpha = 0

    for i in text:
        if (i.isalpha()) == True:
            alpha += 1

    return alpha


def count_words(text):
    # function takes text from user and counts words in text

    count = 1
    n = len(text)

    for i in range(n):
        if (text[i].isspace()):
            count += 1
    # print(words)
    return count


def count_sent(text):
    # function counts sentences using regex library

    sents = len(re.split(r'[.!?]+', text))

    # print(sents)
    return sents - 1


main()